### ISSUE TITLE
[feat] Adding underlines to inline links

### ISSUE THREAD

**Issue by marionnegp** [RedHat-UX/red-hat-design-system#2056]
### Description

Our inline links should be underlined to meet accessibility requirements.

- [ ] Audit elements and patterns that need underlined inline links
- [ ] Update Figma library
- [ ] Update code
- [ ] Update docs

### Suggested solution

Use the styling in this [CodePen](https://codepen.io/dcaryll/pen/LYwRVqX). 

Additional info and considerations for RHDS in [this PowerPoint](https://docs.google.com/presentation/d/1x2z4hYZ4UarWdCvEYDKc7_MTffFeeuJwNBi1iHO4oOk/edit#slide=id.g2e270e37879_0_20)

### Additional context

This may close #747.

**Comment by hellogreg** [comment]
Let me know if you want me to to tackle anything regarding the docs for this: either writing or editing.

**Comment by marionnegp** [comment]
These changes should resolve issue #747

**Comment by marionnegp** [comment]
@coreyvickery @markcaron @hellogreg, since inline link styling is going under the new Foundations > Interactions page, here's a [mockup of the new page](https://www.figma.com/design/hV51MV4NwYbLCQ4J0UaSqk/Foundations---Interactions?node-id=1-3) that Corey helped outline. Right now, we have content for only the "Overview" and "Link" subpages. I'm hoping to get this out while we work on the other content so that we can also release the link underline styles in Figma.

To avoid duplicating content, I think we're removing the "Links" section from Foundations > Color > Accessibility. We're also updating the "Links" section under Accessibility > Design, which is mocked up in [this file](https://www.figma.com/design/HxNA3yKXmrWpOrDtQe8bJv/Accessibility---Design?node-id=2001-2486). (Note that this last Figma link shows only the section that's getting updated. I don't think we had a mockup for the Accessibility pages in Figma.)

**Comment by marionnegp** [comment]
@markcaron @coreyvickery @hellogreg, here's a link to the updated [mockup](https://www.figma.com/design/hV51MV4NwYbLCQ4J0UaSqk/Foundations---Interactions?node-id=1-3) for Foundations > Interactions > Links page!
